client/wfnews-war/src/main/angular/src/app/components/report-of-fire/compass-page/rof-compass-page.component.ts

```diff
@@ -98,17 +103,19 @@ handler(e) {
     if (document.getElementById("compass-face-image")) document.getElementById("compass-face-image").style.transform = `rotate(${-compassHeading}deg)`;
     if (document.getElementById("compass-heading")) document.getElementById("compass-heading").innerText = compassHeading.toString() + "° " + cardinalDirection;
 
-    this.reportOfFire.compassHeading = compassHeading;
+    self.reportOfFire.compassHeading = compassHeading;
 
-    this.useMyCurrentLocation();
+    self.useMyCurrentLocation(self);
```

Comment:
if `self` is calling `useMyCurrentLocation`, why do you need to pass `self` into it? It should already be accessible via `this`